detox/src/errors/longreads/failedToReachTheApp.js

```diff
@@ -0,0 +1,58 @@
+const DetoxRuntimeError = require('../DetoxRuntimeError');
+
+const message = 'Failed to reach the app over the web socket connection.';
+
+const hint1_B = `\
+1. If you don't see your app running on the device, there's a chance
+   that your app has crashed prematurely. To get the crash details,
+   you can run Detox tests with "--record-logs all" CLI option
+   and then inspect the device logs in the artifacts folder.\
+`;
+
+const hint1_A = `\
+Have you forgotten to write 'device.launchApp()' at the beginning
+of your test? If that's not the case, see the other options.
+
+${hint1_B}
+`;
+
+const hint2 = `\
+2. If your app IS running on the device, yet you see this message:
+a) The native part of Detox failed to connect to the Detox server over
+   web sockets. If this is the case, the device's logs should contain
+   messages about those failed connection attempts.
+
+b) The app is running without Detox native code injected.
```

Comment:
`You've either launched the app manually, or there was a device-side error related to Detox... (etc.)`
(the point is to avoid getting into technical, architecture details such as Detox native code, injection, etc.)